dhcpv4/client4/client_darwin.go

```diff
@@ -0,0 +1,3 @@
+// Package client4 is deprecated. Use "nclient4" instead.
+package client4
```

Comment:
I would advise against making an empty package file just to make a package build. A package that is not supported on darwin at all should just not build on darwin. If you don't plan to use client4 and you use nclient4 instead, just leave it unsupported and not building.

Reply:
Okay, a function that just returns unsupported is also undesired... The 2 desired states from a user point of view are: it either does not compile (compiler says unsupported!) or it compiles and has full support. Why are you trying to make client4 compile if you're not trying to use it? I think just making nclient4 compile is a good enough goal if that's what you're trying to use?

Reply:
Compiling but having no functions and compiling but only returning errors are silent failures from a user perspective. It's like Python - it can be interpreted, but you don't know what's wrong until you run it. I'd prefer if we remain in the "not compilable" state for things that we do not support. I guess "compiling but having no functions" is slightly better than "compiling but only returning errors" from that perspective, but I still don't get why you're trying to do it.

Reply:
yeah. this is a hack. I'll let somebody else work out how to do this. This package is wearying.